### PR TITLE
Add time to callback

### PR DESCRIPTION
**Pull request by psc-g**
Add time as an optional parameter to callback.

adarob@ had suggested this in the original pull request that added the callback but I said it wasn't necessary at the time. Turns out I do need it for a new thing I'm building :).
Making it optional so it's backwards compatible.

### CODE REVIEW

File: music/src/core/player.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ function compareQuantizedNotes(a: NoteSequence.INote, b: NoteSequence.INote) {
 export abstract class BasePlayerCallback {
   /* Will be called for each time/note pair in a sequence being played.
    */
-  abstract run(n: NoteSequence.INote): void;
+  abstract run(n: NoteSequence.INote, t?: number): void;
```
Comment (adarob) [inline music/src/core/player.ts]: Add parameters to docstring.

Comment (psc-g) [reply music/src/core/player.ts]: argh, sloppy of me, sorry. done.

File: music/src/core/player.ts [hunk 2]
```diff
@@ -47,9 +47,14 @@ function compareQuantizedNotes(a: NoteSequence.INote, b: NoteSequence.INote) {
...
  */
 export abstract class BasePlayerCallback {
-  /* Will be called for each time/note pair in a sequence being played.
+  /**
+   *   Will be called for each time/note pair in a sequence being played.
+   *
+   *   @param n A NoteSequence.INote object, which is the note being played
+   *     at the moment.
+   *   @param t A number, the time at which the note is being played.
    */
```
Comment (adarob) [inline music/src/core/player.ts]: No need to specify the type since it is specified below and added to generated documentation automatically.

Comment (psc-g) [reply music/src/core/player.ts]: done.

File: music/src/core/player.ts [hunk 3]
```diff
@@ -47,9 +47,13 @@ function compareQuantizedNotes(a: NoteSequence.INote, b: NoteSequence.INote) {
  * played.
  */
 export abstract class BasePlayerCallback {
-  /* Will be called for each time/note pair in a sequence being played.
+  /**
+   *   Will be called for each time/note pair in a sequence being played.
```
Comment (adarob) [inline music/src/core/player.ts]: Sorry to nit, but there should just be a single space after the *.

Comment (psc-g) [reply music/src/core/player.ts]: nit away! done.
i was basing myself on the spacing used in the BasePlayer constructor docstring.

**Comment by psc-g** [comment]
friendly ping :)

**Pull request merged**